include/frame.h

```diff
@@ -17,20 +17,42 @@
 #include <vector>
 
 #include "refcount.h"
+#include "messages_generated.h"
 
 #ifdef _MSC_VER
 typedef unsigned int uint32_t;
 typedef unsigned short uint16_t;
 typedef int int32_t;
 #endif
 
-// TODO Check types !
-
 namespace torchcraft {
 namespace replayer {
+
+struct Unit;
+struct Resources;
+struct Bullet;
+struct Action;
+class Frame;
+class FrameDiff;
+namespace detail { class UnitDiff; }
+
+std::ostream& operator<<(std::ostream& out, const Frame& o);
+std::istream& operator>>(std::istream& in, Frame& o);
+std::ostream& operator<<(std::ostream& out, const FrameDiff& o);
+std::istream& operator>>(std::istream& in, FrameDiff& o);
+
+struct Bullet {
+  int32_t type, x, y;
+};
+
+struct Action { // corresponds to a torchcraft message
```

Comment:
No, it doesn't actually. It was up to the client to decide what this means IIRC. @ebetica I guess we could remove this as well while doing #233 ?